tests/unit/modules/test_asset.py

```diff
@@ -13,25 +14,242 @@
+                    }
+                ]
+            },
+            {
+                "name": "asset",
+                "builds": [
+                    {
+                        "url": "http://def.com",
+                        "sha512": "abc",
+                    },
+                    {
+                        "url": "http://def.com",
+                        "sha512": "abc",
+                        "filters": ["abc == def"],
+                    }
+                ]
+            },
+        ) is True
+
+
+class TestDoBuildsDiffer:
+    def test_equal_with_different_order(self):
+        assert asset._do_builds_differ(
+            [
+                {"sha512": "b", "url": "b"},
+                {"sha512": "a", "url": "a"},
+            ],
+            [
+                {"sha512": "a", "url": "a"},
+                {"sha512": "b", "url": "b"}
+            ],
+        ) is False
+
+    def test_equal_with_different_order_of_filters(self):
+        assert asset._do_builds_differ(
+            [{
+                "sha512": "a",
+                "url": "a",
+                "filters": ["a", "b", "c"],
+            }],
+            [{
+                "sha512": "a",
+                "url": "a",
+                "filters": ["c", "a", "b"],
+            }],
+        ) is False
+
+    def test_equal_with_none_values(self):
+        assert asset._do_builds_differ(
+            [{
+                "sha512": "a",
+                "url": "a",
+                "filters": None
+            }],
+            [{
+                "sha512": "a",
+                "url": "a",
+                "headers": None,
+            }],
+        ) is False
+
+    def test_equal(self):
+        assert asset._do_builds_differ(
+            [
+                {
+                    "url": "a",
+                    "sha512": "a",
+                    "filters": ["a", "b", "c"],
+                    "headers": {
+                        "foo": "bar",
+                        "bar": "foo",
+                    },
+                },
+                {
+                    "url": "b",
+                    "sha512": "b",
+                    "filters": ["d", "e", "f"],
+                    "headers": {
+                        "abc": "abc",
+                        "def": "def",
+                    },
+                },
+            ],
+            [
+                {
+                    "url": "b",
+                    "sha512": "b",
+                    "filters": ["f", "e", "d"],
+                    "headers": {
+                        "def": "def",
+                        "abc": "abc",
+                    },
+                },
+                {
+                    "url": "a",
+                    "sha512": "a",
+                    "filters": ["b", "c", "a"],
+                    "headers": {
+                        "bar": "foo",
+                        "foo": "bar",
+                    },
+                },
+            ],
+        ) is False
```

Comment:
This scenario is also almost covered. If we add a few more entries to the `headers` dictionary in the `TestDoDiffer.test_equal` case, we can drop this one.